Create tdot.md

shengda opened the pull request:
### Project Abstract

This is a follow-up of the [stable asset grant](https://github.com/w3f/Grants-Program/pull/286).

tDOT is a DOT-pegged derivative built on top of the stable asset protocol. It aims at generating unified DOT liquidity across Polkadot parachains and applications.

### For which grant [level](https://github.com/w3f/Grants-Program#level_slider-levels) are you applying? 
- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [x] **Level 2**:  Up to $50,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for > $100k Web3 Foundation Council approval)

### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied, renamed ( `project_name.md`) and updated.
- [x] A BTC or Ethereum (DAI/USDT) address for the payment of the milestones is provided inside the application.
- [x] I have read and acknowledged the [terms and conditions](https://github.com/w3f/Grants-Program/blob/master/docs/T&Cs.md).
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process).

alxs commented:
Thanks for the application @shengda. Could you also fill in the project abstract? Also as far as I understand it you're basically adding cross-chain integration to an existing pallet and to be honest $20k seems a little high to me for this. Feel free to argue otherwise.

alxs commented:
On hold pending review of the [last Stable Asset milestone](https://github.com/w3f/Grant-Milestone-Delivery/pull/374).

shengda commented:
Project abstract added. The project is more than simply adding cross-chain integrations. For example, the hosting parachain needs to control minter access and minter cap across other parachains. The minting/redeeming/swap flows are also more complicated when cross-chain integration. That's why we set up a $20k budget for the grant.

shengda commented:
Hi @Noc2 , thank you for your comments.

* This is not a parachain-specific solution. There will be a single hosting minter and multiple remote minters, all of them can be deployed in any parachain. In fact, our original plan is to deploy the hosting minter in Statemine as a smart contract so that any other parachain can deploy remote minters. Since Statemine does not have smart contract functionalities yet, and Acala has both stake and crowdloan DOT available, we decide to launch our hosting minter in Acala. It's still possible for us to migrate hte hosting minter to another parachain, or a dedicated parachain in the future;

* Cross-chain minting is relatively simple. Remote minter B on parachain B manages their native liquidity and sends XCM messages on hosting minter A on parachain A to mint tDOT. tDOT is minted on parachain A since it's the hosting parachain. Cross-chain burning will be more complicated. Since tDOT must be burned on parachain A and the underlying asset will be released on parachain B, we need to make sure that redeemer can still get expected outcome with specified slippage. Our current solution is:
     1. Lock tDOT to redeem on parachain A;
     2. Parachain A sends an XCM message to parachain B to redeem to specific assets with expected slippage;
     3. If parachain B can redeem within slippage, it releases the underlying asset and sends another XCM mesage back to parachain A to burn the locked tDOT;
     4. If parachain B cannot redeem within slippage, it sends back another XCM message back to parachain A to unlock tDOT.
Crossc-chain swap is similar to cross-chain redeem but will be slightly more complicated. tDOT will use XCM v3;

* Attack vector includes both attack on individual pools and between pools. For example, attackers can choose to attack one single underlying asset, or the correlation between two underlying assets in different pools. This will be addressed by different pool parameters and governance, which is outside the scope of this grant.

Noc2 reviewed (changes requested):
Thanks for the quick reply. In this case, could you try to make the wording in your application more generic or simply remove acala for now? We usually don’t care where you deploy your application and we don’t want to cause any confusion with the application. So it doesn’t look like we support a specific parachain. Could you also add the description regarding XCM to the deliverables, potentially also benchmarking and any additional functionality? The reason I’m asking is because in its current form the deliverables could be relatively easily fulfilled by just implementing functions, which you mention there. So given the price, it would be nice if you could add additional functionality. This could also be a research paper about the security assumptions and problems. For example, one parachain can not be sure that the code executes correctly on the other parachain. https://wiki.polkadot.network/docs/learn-spree

shengda commented:
@Noc2 I have removed deployment information and added more details on the deliverables. Please review.

Noc2 reviewed (dismissed):
Thanks for the update. I still think that this application is relatively expensive, but I’m willing to go ahead with it and share it with the rest of the team.

shengda commented:
@Noc2 I review the milestone and update the cost.  The cost is not an important consideration for us; we would like to move forward as fast as possible.

Noc2 reviewed (approved):
Thanks a lot for the update. I’m still happy to go ahead with it ;-)

takahser commented on applications/tdot.md:
```diff
@@ -0,0 +1,168 @@
+
+## Project Overview :page_facing_up:
+
+This application is a follow-up of the Stable Asset grant.
```
I think it'd be useful to link that grant here.

takahser reviewed (changes requested):
@shengda thanks for your application, this looks quite well already. 👍

Just one thing: It'd be nice if the milestones would adhere to our [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md#development-roadmap-nut_and_bolt):

> Deliverables 0a-0d are mandatory for all milestones, and deliverable 0e at least for the last one

- As far as I can tell, your application already contains the equivalent of 0a-0d but the deliverable numbers are different. So fixing this would be limited to the restructuring of the deliverables within the milestones.
- 0e is currently missing. Mostly, teams add it to the last milestone within a grants application.

shengda commented:
> @shengda thanks for your application, this looks quite well already. 👍
> 
> Just one thing: It'd be nice if the milestones would adhere to our [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md#development-roadmap-nut_and_bolt):
> 
> > Deliverables 0a-0d are mandatory for all milestones, and deliverable 0e at least for the last one
> 
> * As far as I can tell, your application already contains the equivalent of 0a-0d but the deliverable numbers are different. So fixing this would be limited to the restructuring of the deliverables within the milestones.
> * 0e is currently missing. Mostly, teams add it to the last milestone within a grants application.

@takahser I have updated the deliverable numbers and also added two articles as deliverables.

takahser reviewed (approved):
@shengda thanks for the update, happy to go along with it as well. 👍
Looking forward to your delivery!

keeganquigley commented:
Hi @shengda just curious if you are able to give an update on the progress of milestone 4?

keeganquigley commented:
@shengda friendly reminder.

keeganquigley commented:
Hi @shengda please note that if we don't hear back from you in 2 weeks' time we will terminate the grant due to inactivity. Let us know if you have any questions.
